diff --git a/scripts/mkcloud b/scripts/mkcloud
index 637e9212..59be0593 100755
--- a/scripts/mkcloud
+++ b/scripts/mkcloud
@@ -787,6 +787,15 @@ function instnodes
     echo "Waiting for the installation of the nodes ..."
     safely onadmin waitcloud
     safely onadmin post_allocate
+    onadmin_post_allocated=1
+}
+
+function post_allocate
+{
+    if [[ $onadmin_post_allocated != 1 ]] ; then
+        safely onadmin post_allocate
+        onadmin_post_allocated=1
+    fi
 }

 function deployexternalceph
@@ -1474,7 +1483,7 @@ allcmds="$step_aliases _test_setuphost \
     all all_noreboot all_batch all_batch_noreboot instonly \
     plain plain_with_upgrade cleanup setuphost prepare libvirt_prepare setupadmin \
     prepareinstcrowbar bootstrapcrowbar instcrowbar instcrowbarfromgit setupnodes \
-    setupcompute instnodes instcompute proposal testsetup rebootcrowbar \
+    setupcompute instnodes post_allocate instcompute proposal testsetup rebootcrowbar \
     rebootcloud addupdaterepo runupdate testupdate \
     crowbarbackup crowbarrestore shutdowncloud restartcloud qa_test help \
     rebootneutron cloudupgrade \